### PR TITLE
Feature/999 update single attr value by eid

### PR DESCRIPTION
**Pull request by kzangeli**
Implements #999

PUT /v2/entities/\<entityId\>/attrs/\<attrName\>/value

### CODE REVIEW

File: test/functionalTest/cases/0991_update_single_attr_by_eid/update_single_attr_by_eid.test [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+--NAME--
+PUT Individual Entity Attribute
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+
+--SHELL--
+
+#
```
Comment (fgalan) [inline test/functionalTest/cases/0991_update_single_attr_by_eid/update_single_attr_by_eid.test]: I understand all these steps are related with changing attribute value. Some test to check updating attribute NGSI type and metadata should be also added (but in a different .test).

In addition, a test to check that, in a entity with 2 attributes, the operaiton only affects to the right one (leaving the other untouched) should be added.

Comment (kzangeli) [inline test/functionalTest/cases/0991_update_single_attr_by_eid/update_single_attr_by_eid.test]: Added a test of "leaving the other untouched" in 3cbf771 (also, I changed the behavior of the operation from REPLACE to UPDATE as this is not a normal PUT - it's a 'concentrated' PUT ...)

Comment (kzangeli) [inline test/functionalTest/cases/0991_update_single_attr_by_eid/update_single_attr_by_eid.test]: PR changed to resolve issue #999 and not #991.
I believe all necessary functests are present

Fixed in 3cbf771, cd9c877 and 361cbb6

File: test/functionalTest/cases/0999_update_single_attr_value_by_eid/update_single_attr_value_by_eid_error_handling.test [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+Error handling For PUT Individual Entity Attribute
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+
+--SHELL--
+
+#
+# 01. PUTT /v2/entities/E1/attrs/A1/value to see BAD METHOD, only GET, PUT
```
Comment (fgalan) [inline test/functionalTest/cases/0999_update_single_attr_value_by_eid/update_single_attr_value_by_eid_error_handling.test]: Typo: PUTT -> PUT

File: test/functionalTest/cases/0999_update_single_attr_value_by_eid/update_single_attr_value_by_eid_error_handling.test [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+
+--SHELL--
+
+#
+# 01. PUTT /v2/entities/E1/attrs/A1/value to see BAD METHOD, only GET, PUT 
+# 02. PUT /v2/entities/E1/attrs/A1/value without 'value' in payload
+# 03. PUT /v2/entities/E1/attrs/A1/value with 'value' and 'value2' in payload
```
Comment (fgalan) [inline test/functionalTest/cases/0999_update_single_attr_value_by_eid/update_single_attr_value_by_eid_error_handling.test]: Testing with 'value' and 'type' and get an error would be also interesting.

File: test/functionalTest/cases/0999_update_single_attr_value_by_eid/update_single_attr_value_by_eid_error_handling.test (hunk 2 above)
Comment (kzangeli) [inline test/functionalTest/cases/0999_update_single_attr_value_by_eid/update_single_attr_value_by_eid_error_handling.test]: Actually no, not a typo.
The test here is to provoke "BAD METHOD"

File: test/functionalTest/cases/0999_update_single_attr_value_by_eid/update_single_attr_value_by_eid_error_handling.test (hunk 3 above)
Comment (kzangeli) [inline test/functionalTest/cases/0999_update_single_attr_value_by_eid/update_single_attr_value_by_eid_error_handling.test]: Sure, it's the same though - everything but 'value' gives an error.
However, 'type' is more interesting than 'asdasd' or 'value2'.

Comment (kzangeli) [inline test/functionalTest/cases/0999_update_single_attr_value_by_eid/update_single_attr_value_by_eid_error_handling.test]: Fixed in 3b814c1

File: test/functionalTest/cases/0999_update_single_attr_value_by_eid/update_single_attr_value_by_eid_error_handling.test (hunk 2 above)
Comment (fgalan) [inline test/functionalTest/cases/0999_update_single_attr_value_by_eid/update_single_attr_value_by_eid_error_handling.test]: Maybe a more explicit token should be used, such as NOVERB or BADVERD.

Comment (kzangeli) [inline test/functionalTest/cases/0999_update_single_attr_value_by_eid/update_single_attr_value_by_eid_error_handling.test]: BADVERB used in 99569c1

**Comment by fgalan** [comment]
LGTM

**Pull request merged**